---
title: "Therm. var. microbes - ITS2 analysis"
author: "Nicola G. Kriefall"
date: "09/09/22"
output:
 rmarkdown::html_document:
    theme: cerulean
    toc: yes
    toc_float: yes
    highlight: haddock
    number_sections: true
---

# SymPortal pre-submission

Submitting ITS2 samples to [Symportal](symportal.org)

Prior to submission: retaining only paired reads that match ITS2 primer (we pooled with 16S during sequencing)

```{bash setup symportal, eval=FALSE}
# *note: most of this was written by Dr. Carly D. Kenkel

# in Terminal home directory:
# following instructions of installing BBtools from https://jgi.doe.gov/data-and-tools/bbtools/bb-tools-user-guide/installation-guide/
# 1. download BBMap package, sftp to installation directory
# 2. untar: 
tar -xvzf BBMap_(version).tar.gz
# 3. test package:
cd bbmap
~/bin/bbmap/stats.sh in=~/bin/bbmap/resources/phix174_ill.ref.fa.gz

# primers for ITS2:
#fwd:
#GAATTGCAGAACTCCGTGAACC
#rev:
#CGGGTTCWCTTGTYTGACTTCATGC

# making a sample list based on the first phrase before the underscore in the .fastq name
ls *R1_001.fastq | cut -d '_' -f 2 > samples.list

# cuts off the extra words in the .fastq files
for file in $(cat samples.list); do mv *${file}*R1_001.fastq ${file}_R1.fastq; mv *${file}*R2_001.fastq ${file}_R2.fastq; done

# gets rid of reads that still have the adaptor sequence
# [I stopped doing this 'cause it never happened]
#for file in $(cat samples.list); do ~/bin/bbmap/bbduk.sh in1=${file}_R1.fastq in2=${file}_R2.fastq ref=adaptors.fasta out1=${file}_R1_NoIll.fastq out2=${file}_R2_NoIll.fastq; done &>bbduk_NoIll.log

##getting rid of first 4 bases (degenerate primers created them)
for file in $(cat samples.list); do ~/bin/bbmap/bbduk.sh in1=${file}_R1.fastq in2=${file}_R2.fastq ftl=4 out1=${file}_R1_No4N.fastq out2=${file}_R2_No4N.fastq; done &>bbduk_No4N.log

# only keeping reads that start with the primer
for file in $(cat samples.list); do ~/bin/bbmap/bbduk.sh in1=${file}_R1_No4N.fastq in2=${file}_R2_No4N.fastq k=15 restrictleft=25 literal=GAATTGCAGAACTCCGTGAACC,CGGGTTCWCTTGTYTGACTTCATGC outm1=${file}_R1_ITS.fastq outu1=${file}_R1_check.fastq outm2=${file}_R2_ITS.fastq outu2=${file}_R2_check.fastq; done &>bbduk_ITS.log
# higher k = more reads removed, but can't surpass k=20 or 21

# renamed them to the shorter version again after checking
for file in $(cat samples.list); do  mv ${file}_*R1*.fastq ${file}_R1.fastq; mv ${file}_*R2*.fastq ${file}_R2.fastq; done 
```

Then transferred all "R1.fastq" & "R2.fastq" files to the folder to be submitted to SymPortal

# SymPortal ITS2 analysis

Information on Symportal output documents [here](https://github.com/didillysquat/SymPortal_framework/wiki/SymPortal-output-documents)

Formatting notes:

- Cleaned up file called 223_20220913T175714_DBV_20220914T011822.profiles.absolute.abund_and_meta.txt to format for phyloseq

## Setup

```{r packages analysis, echo=FALSE}
library(phyloseq)
library('ggplot2')
library('Rmisc')
library(cowplot)
library("ggpubr")
library("vegan")
library(scales)
library(colorspace)
library(funfuns)
```

```{r working directory, include=FALSE}
setwd("/Volumes/GoogleDrive/My Drive/thermvar_microbes_master/thermvar_orbicella/tvm_its2")
```

## Checking for no read samples

```{r counts per sample, eval=F}
counts <- read.csv('tvm_symportal_type_profiles.csv',header=TRUE,row.names=1,check.names=FALSE)
#41 samples, 15 type profiles

sort(rowSums(counts))
counts.no0 <- counts[rowSums(counts)>0,]
#none failed...
```

## Phyloseq object - type profiles

Ran once then saved to re-read in later

```{r phyloseq object type profiles, eval=F}
# import dataframe holding sample information
samdf <- read.csv("tvm_its2_sample_info.csv")
head(samdf)
rownames(samdf) <- samdf$sample_name

# import taxa info
taxa <- read.csv("tvm_symportal_taxa.csv",header=TRUE)
rownames(taxa) <- as.factor(taxa$ITS2_type_profile)
mtaxa <- as.matrix(taxa)

# import counts (absolute abundance from its2 type profiles)
mcounts <- as.matrix(counts)

# Construct phyloseq object 
ps <- phyloseq(otu_table(mcounts, taxa_are_rows=FALSE),
               sample_data(samdf),
               tax_table(mtaxa))

ps
# phyloseq-class experiment-level object
# otu_table()   OTU Table:         [ 15 taxa and 41 samples ]
# sample_data() Sample Data:       [ 41 samples by 6 sample variables ]
# tax_table()   Taxonomy Table:    [ 15 taxa by 4 taxonomic ranks ]

#saveRDS(ps,file="ps.its2.rds")
```

## Plotting

### Packages

```{r}
library(microshades)
```

### Re-read in data

```{r}
ps <- readRDS("ps.its2.RDS")

ps.rel <- transform_sample_counts(ps, function(x) x / sum(x))
```

### Microshades plot {.tabset}

```{r}
mdf.ps <- prep_mdf(ps.rel,subgroup_level="ITS2_type_profile")

mdf.ps$timepoint <- factor(mdf.ps$timepoint,levels=c("preheat","postheat"))

type_groups <- c(unique(mdf.ps$Majority_ITS2_sequence))

# Create a color object for the specified data
col.mdf.ps <- create_color_dfs(mdf.ps, top_orientation = FALSE,group_level="Majority_ITS2_sequence",subgroup_level="ITS2_type_profile",selected_groups=c("D1","B1","B2","A3"),cvd=TRUE)

#Extract
col.mdf.ps.m <- col.mdf.ps$mdf
col.mdf.ps.c <- col.mdf.ps$cdf

col.mdf.ps.c.new <- color_reassign(col.mdf.ps.c,
                          group_assignment = c("D1","B1","B2","A3"),
                          color_assignment = c("micro_cvd_purple","micro_cvd_turquoise","micro_cvd_green","micro_cvd_orange"),group_level="Majority_ITS2_sequence")

#both
ms.plot.both <- plot_microshades(col.mdf.ps.m, col.mdf.ps.c.new,x="timepoint")+
  facet_wrap(host_species~genotype*treatment,scales="free")+
  theme_cowplot()+
  theme(axis.text.x=element_text(angle=45,vjust=0.75,hjust=0.75))+
  guides(color=guide_legend("Majority seq.-type profile"),fill=guide_legend("Majority seq.-type profile"))
```

#### Favs type profile - all

```{r}
col.mdf.ps.m.fav <- subset(col.mdf.ps.m,host_species=="faveolata")

##control
col.mdf.ps.fav.m.con <- subset(col.mdf.ps.m.fav,treatment=="control")
col.mdf.ps.fav.m.con$treatment <- gsub("control", "Control",col.mdf.ps.fav.m.con$treatment)

ms.plot.con <- plot_microshades(col.mdf.ps.fav.m.con, col.mdf.ps.c.new,x="timepoint")+
  facet_grid(treatment~genotype,scales="free")+
  theme_cowplot()+
  theme(axis.text.x=element_text(angle=45,hjust=0.75))+
  guides(color=guide_legend("Majority seq.-type profile"),fill=guide_legend("Majority seq.-type profile"))+
  scale_x_discrete(labels=c("Pre","Heat"))+
  xlab("")
ms.plot.con

##therm var
col.mdf.ps.fav.m.var <- subset(col.mdf.ps.m.fav,treatment=="thermvar")
col.mdf.ps.fav.m.var$treatment <- gsub("thermvar", "Therm var",col.mdf.ps.fav.m.var$treatment)

ms.plot.var <- plot_microshades(col.mdf.ps.fav.m.var, col.mdf.ps.c.new,x="timepoint")+
  facet_grid(treatment~genotype,scales="free")+
  theme_cowplot()+
  theme(axis.text.x=element_text(angle=45,hjust=0.75))+
  guides(color=guide_legend("Majority seq.-type profile"),fill=guide_legend("Majority seq.-type profile"))+
  scale_x_discrete(labels=c("Pre","Heat"))+
  xlab("")
ms.plot.var

gg.bar.fav <- ggarrange(ms.plot.con,ms.plot.var,nrow=2, legend="none",align="hv")
gg.bar.fav2 <- annotate_figure(gg.bar.fav, top=text_grob("Orbicella faveolata",face="italic",size=16))

#ggsave("gg.its2.fav.types.pdf",width=10)
```

#### Franks type profile - all

```{r}
col.mdf.ps.m.fra <- subset(col.mdf.ps.m,host_species=="franksi")
#extra re-done sample
col.mdf.ps.m.fra <- subset(col.mdf.ps.m.fra,Sample!="KA4-2")

##control
col.mdf.ps.fra.m.con <- subset(col.mdf.ps.m.fra,treatment=="control")
col.mdf.ps.fra.m.con$treatment <- gsub("control", "Control",col.mdf.ps.fra.m.con$treatment)

ms.plot.con <- plot_microshades(col.mdf.ps.fra.m.con, col.mdf.ps.c.new,x="timepoint")+
  facet_grid(treatment~genotype,scales="free")+
  theme_cowplot()+
  theme(axis.text.x=element_text(angle=45,hjust=0.75))+
  guides(color=guide_legend("Majority seq.-type profile"),fill=guide_legend("Majority seq.-type profile"))+
  scale_x_discrete(labels=c("Pre","Heat"))+
  xlab("")+
  ylab("")
ms.plot.con

##therm var
col.mdf.ps.fra.m.var <- subset(col.mdf.ps.m.fra,treatment=="thermvar")
col.mdf.ps.fra.m.var$treatment <- gsub("thermvar", "Therm var",col.mdf.ps.fra.m.var$treatment)

ms.plot.var <- plot_microshades(col.mdf.ps.fra.m.var, col.mdf.ps.c.new,x="timepoint")+
  facet_grid(treatment~genotype,scales="free")+
  theme_cowplot()+
  theme(axis.text.x=element_text(angle=45,hjust=0.75))+
  guides(color=guide_legend("Majority seq.-type profile"),fill=guide_legend("Majority seq.-type profile"))+
  scale_x_discrete(labels=c("Pre","Heat"))+
  xlab("")+
  ylab("")
ms.plot.var

gg.bar.fra <- ggarrange(ms.plot.con,ms.plot.var,nrow=2, legend="none",align="hv")
gg.bar.fra2 <- annotate_figure(gg.bar.fra, top=text_grob("Orbicella franksi",face="italic",size=16))

#ggsave("gg.its2.fra.types.pdf",width=10)
```

### Both species plot

```{r}
#just legend
# Extract the legend. Returns a gtable
leg <- get_legend(ms.plot.both)

ggarrange(gg.bar.fav2,gg.bar.fra2,as_ggplot(leg),ncol=3,align="hv",labels=c("(a)","(b)",""))
ggsave("gg.its2.bothspp.pdf",width=10,height=4.5)
```

### Different plot function to remove black bars

```{r ps plot2 fxn}
plot_bar2 <-  function (physeq, x = "Sample", y = "Abundance", fill = NULL, title = NULL, facet_grid = NULL, border_color = NA) 
{
  mdf = psmelt(physeq)
  p = ggplot(mdf, aes_string(x = x, y = y, fill = fill))
  p = p + geom_bar(stat = "identity", position = "stack",  color = border_color)
  p = p + theme(axis.text.x = element_text(angle = -90, hjust = 0))
  if (!is.null(facet_grid)) {
    p <- p + facet_grid(facet_grid)
  }
  if (!is.null(title)) {
    p <- p + ggtitle(title)
  }
  return(p)
}
```

### The plots{.tabset}

#### Franksi

```{r}
plot_bar2(ps.fra, fill="ITS2_type_profile")+
    #ggtitle("Pre")+
    theme_cowplot()+
    #theme(axis.text.x=element_text(angle=90,hjust=1))+
    xlab("")+
  facet_wrap(treatment~timepoint,scales="free_x")+
  #scale_x_discrete(labels=c("Inshore","Offshore"))+
  #scale_fill_manual(values=maj.seq.colors)+
  geom_bar(stat = "identity")
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
  #theme(axis.text.x=element_text(angle=45))
```

```{r}
plot_bar2(ps.fav, fill="ITS2_type_profile",x="genotype")+
    #ggtitle("Pre")+
    theme_cowplot()+
    #theme(axis.text.x=element_text(angle=90,hjust=1))+
    xlab("")+
  facet_wrap(treatment~timepoint,scales="free_x")+
  #scale_x_discrete(labels=c("Inshore","Offshore"))+
  #scale_fill_manual(values=maj.seq.colors)+
  geom_bar(stat = "identity")
  #theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
  #theme(axis.text.x=element_text(angle=45))
  

```

## Bubble plot?

```{r}
library(dplyr)

sppAbun <- as.data.frame(ps@otu_table)

sppAbun[sppAbun > 0] <- 1 #converts from abundance to P/A
sppAbun

rowSums(sppAbun)

ps@otu_table <- otu_table(sppAbun,taxa_are_rows=F)
ps.pa1 <- psmelt(ps)
ps.pa <- subset(ps.pa1,Sample!="KA4-2")

ps.pa.grp <- ps.pa %>%
  group_by(host_species,treatment,timepoint,Sample) %>%
  summarize_at('Abundance',sum)

ps.pa.grp.n <- ps.pa.grp %>%
  group_by(host_species,treatment,timepoint,Abundance) %>%
  summarise(n=n())

ps.pa.grp.n$timepoint <- factor(ps.pa.grp.n$timepoint,levels=c("preheat","postheat"))

ggplot(ps.pa.grp.n,aes(x=timepoint,y=Abundance,size=n))+
  geom_point()+
  facet_wrap(host_species~treatment)+
  theme_bw()

#ggplot(ps.pa.grp.n,aes(x=timepoint,y=Abundance))+
#  geom_violin()+
#  facet_wrap(host_species~treatment)

```

# PCoA

## All

```{r}
ps.rel.ord <- ordinate(ps.rel,"PCoA",distance="bray")

plot_ordination(ps.rel, ps.rel.ord, color ="timepoint")+
  geom_point(alpha=0.5)+
  #scale_color_manual(values=c("darkslategray3","darkslategray4","#000004"))+
  #scale_shape_manual(values=c(16,15))+
  stat_ellipse()+
  facet_grid(treatment~host_species)+
  theme_cowplot()
```

## Favs

All in one

```{r}
ps.fav <- subset_samples(ps.rel,host_species=="faveolata")

ord.fav <- ordinate(ps.fav, "PCoA", "bray")

plot.fav <- plot_ordination(ps.fav, ord.fav,color="treatment",shape="timepoint")+
  stat_ellipse()+
  scale_color_manual(values=c("#B4BD93","#C7522B"))+
  #scale_fill_manual(values=c("#B4BD93","#C7522B"))+
  scale_shape_manual(values=c(21,24))+
  #scale_fill_manual(values=c("white","gray"))+
  theme_cowplot()
plot.fav

plot.fav.data <- plot.fav[["data"]]

ggplot(plot.fav.data,aes(x=Axis.1,y=Axis.2,shape=treat_time,fill=treat_time,linetype=timepoint))+
  geom_point()+
  stat_ellipse(aes(linetype=timepoint))+
  scale_fill_manual(name="Treat",values=c("#ECA313","#0085A7","#ECA313","#0085A7"),labels=c("Ctrl heat","Ctrl pre","TV heat","TV pre"))+
  scale_shape_manual(name="Treat",values=c(21,21,24,24),labels=c("Ctrl heat","Ctrl pre","TV heat","TV pre"))+
  scale_linetype_manual(name="Time point",values=c("dashed","dotted"),labels=c("Heat","Pre-heat"))+
  #scale_color_manual(name="Treat",values=c("ECA313","#0085A7"))+
  ggtitle("Rel. - Ofav")+
  xlab("Axis 1 (22.2%)")+
  ylab("Axis 2 (15.4%)")+
  theme_cowplot()

#ggsave("its2.pca.fav",height=2.5,width=3.5)

#cvdPlot(plot.fav.data)
```

```{r}
ps.fav <- subset_samples(ps.rel,host_species=="faveolata")
ps.fav.pre <- subset_samples(ps.rel,timepoint=="preheat")
ps.fav.pos <- subset_samples(ps.rel,timepoint=="postheat")

ps.fav.pre.ord <- ordinate(ps.fav.pre,"PCoA",distance="bray")

plot_ordination(ps.fav.pre, ps.fav.pre.ord, color ="treatment")+
  geom_point(alpha=0.5)+
  stat_ellipse(aes(color=treatment,linetype=treatment))+
  scale_color_manual(name="Treatment",values=c("#B9C184","#D66982"),labels=c("Ctrl","TV"))+
  scale_shape_manual(name="Treatment",values=c(16,17),labels=c("Ctrl","TV"))+
  scale_linetype_manual(name="Treatment",values=c("solid","dashed"),labels=c("Ctrl","TV"))+
  facet_grid(~timepoint)+
  theme_cowplot()

ps.fav.pos.ord <- ordinate(ps.fav.pos,"PCoA",distance="bray")

plot_ordination(ps.fav.pos, ps.fav.pos.ord, color ="treatment")+
  geom_point(alpha=0.5)+
  stat_ellipse(aes(color=treatment,linetype=treatment))+
  scale_color_manual(name="Treatment",values=c("#B9C184","#D66982"),labels=c("Ctrl","TV"))+
  scale_shape_manual(name="Treatment",values=c(16,17),labels=c("Ctrl","TV"))+
  scale_linetype_manual(name="Treatment",values=c("solid","dashed"),labels=c("Ctrl","TV"))+
  facet_grid(~timepoint)+
  theme_cowplot()

ps.fav.con <- subset_samples(ps.fav,treatment=="control")
ps.fav.con.ord <- ordinate(ps.fav.con,"PCoA",distance="bray")

plot_ordination(ps.fav.con, ps.fav.con.ord, color ="timepoint")+
  geom_point(alpha=0.5)+
  stat_ellipse()+
  #scale_color_manual(name="Treatment",values=c("#B9C184","#D66982"),labels=c("Ctrl","TV"))+
  #scale_shape_manual(name="Treatment",values=c(16,17),labels=c("Ctrl","TV"))+
  #scale_linetype_manual(name="Treatment",values=c("solid","dashed"),labels=c("Ctrl","TV"))+
  facet_grid(~treatment)+
  theme_cowplot()

ps.fav.var <- subset_samples(ps.fav,treatment=="thermvar")
ps.fav.var.ord <- ordinate(ps.fav.var,"PCoA",distance="bray")

namez <- c(`thermvar` = "Therm. var.")

plot_ordination(ps.fav.var, ps.fav.var.ord, color ="timepoint",shape="treatment")+
  geom_point(alpha=0.5)+
  stat_ellipse()+
  scale_color_manual(name="Time point",values=c("#0085A7","#ECA313"),labels=c("Pre-heat","Heat"))+
  scale_shape_manual(values=17)+
  #scale_linetype_manual(name="Treatment",values=c("solid","dashed"),labels=c("Ctrl","TV"))+
  facet_grid(~treatment,labeller = as_labeller(namez))+
  theme_cowplot()+
  guides(shape="none")


```

## Franks

```{r}
ps.fra <- subset_samples(ps.rel,host_species=="franksi")
ps.fra.pre <- subset_samples(ps.rel,timepoint=="preheat")
ps.fra.pos <- subset_samples(ps.rel,timepoint=="postheat")
```

## Statz - nada

```{r}
##all all
seq.rel <- data.frame(ps.rel@otu_table)
samdf.rel <- data.frame(ps.rel@sam_data)

dist.rel <- vegdist(seq.rel, method="bray")

bet.rel <- betadisper(dist.rel,samdf.rel$treatment)
permutest(bet.rel,permutations=999)
plot(bet.rel) #ns
bet.rel <- betadisper(dist.rel,samdf.rel$timepoint)
permutest(bet.rel,permutations=999)
plot(bet.rel) #ns

adonis2(dist.rel ~ treatment*timepoint, data=samdf.rel, permutations=999) #ns
```

```{r}
##all
seq.fav <- data.frame(ps.fav@otu_table)
samdf.fav <- data.frame(ps.fav@sam_data)

dist.fav <- vegdist(seq.fav, method="bray")

bet.fav <- betadisper(dist.fav,samdf.fav$treatment)
permutest(bet.fav,permutations=999)
plot(bet.fav) #ns
bet.fav <- betadisper(dist.fav,samdf.fav$timepoint)
permutest(bet.fav,permutations=999)
plot(bet.fav) #ns

adonis2(dist.fav ~ treatment*timepoint, data=samdf.fav, permutations=999) #ns

##by timepoints - pre
seq.fav.pre <- data.frame(ps.fav.pre@otu_table)
samdf.fav.pre <- data.frame(ps.fav.pre@sam_data)

dist.fav.pre <- vegdist(seq.fav.pre, method="bray")

bet.fav.pre <- betadisper(dist.fav.pre,samdf.fav.pre$treatment)
permutest(bet.fav.pre,permutations=999)
plot(bet.fav.pre) #ns

adonis2(dist.fav.pre ~ treatment, data=samdf.fav.pre, permutations=999) #ns

##post
seq.fav.pos <- data.frame(ps.fav.pos@otu_table)
samdf.fav.pos <- data.frame(ps.fav.pos@sam_data)

dist.fav.pos <- vegdist(seq.fav.pos, method="bray")

bet.fav.pos <- betadisper(dist.fav.pos,samdf.fav.pos$treatment)
permutest(bet.fav.pos,permutations=999)
plot(bet.fav.pos) #ns

adonis2(dist.fav.pos ~ treatment, data=samdf.fav.pos, permutations=999) #ns
```

```{r}
##all
seq.fra <- data.frame(ps.fra@otu_table)
samdf.fra <- data.frame(ps.fra@sam_data)

dist.fra <- vegdist(seq.fra, method="bray")

bet.fra <- betadisper(dist.fra,samdf.fra$treatment)
permutest(bet.fra,permutations=999)
plot(bet.fra) #ns
bet.fra <- betadisper(dist.fra,samdf.fra$timepoint)
permutest(bet.fra,permutations=999)
plot(bet.fra) #ns

adonis2(dist.fra ~ treatment*timepoint, data=samdf.fra, permutations=999) #ns

##by timepoints - pre
seq.fra.pre <- data.frame(ps.fra.pre@otu_table)
samdf.fra.pre <- data.frame(ps.fra.pre@sam_data)

dist.fra.pre <- vegdist(seq.fra.pre, method="bray")

bet.fra.pre <- betadisper(dist.fra.pre,samdf.fra.pre$treatment)
permutest(bet.fra.pre,permutations=999)
plot(bet.fra.pre) #ns

adonis2(dist.fra.pre ~ treatment, data=samdf.fra.pre, permutations=999) #ns

##post
seq.fra.pos <- data.frame(ps.fra.pos@otu_table)
samdf.fra.pos <- data.frame(ps.fra.pos@sam_data)

dist.fra.pos <- vegdist(seq.fra.pos, method="bray")

bet.fra.pos <- betadisper(dist.fra.pos,samdf.fra.pos$treatment)
permutest(bet.fra.pos,permutations=999)
plot(bet.fra.pos) #ns

adonis2(dist.fra.pos ~ treatment, data=samdf.fra.pos, permutations=999) #ns
```

# Post med seqs

Formatting notes:

- Cleaned up file called 223_20220913T175714_DBV_20220914T011822.seqs.absolute.abund_only.txt to format for phyloseq

## Setup

```{r packages analysis, echo=FALSE}
library(phyloseq)
library('ggplot2')
library('Rmisc')
library(cowplot)
library("ggpubr")
library("vegan")
library(scales)
library(colorspace)
library(funfuns)
```

```{r working directory, include=FALSE}
setwd("/Volumes/GoogleDrive/My Drive/thermvar_microbes_master/thermvar_orbicella/tvm_its2")
```

## Checking for no read samples

```{r counts per sample, eval=F}
counts <- read.csv('tvm_symportal_postmedseqs.csv',header=TRUE,row.names=1,check.names=FALSE)
#41 samples, 336 type profiles

sort(rowSums(counts))
counts.no0 <- counts[rowSums(counts)>0,]
#none failed...
```

## Phyloseq object - post med seqs

Ran once then saved to re-read in later

```{r phyloseq object type profiles, eval=F}
# import dataframe holding sample information
samdf <- read.csv("tvm_its2_sample_info.csv")
head(samdf)
rownames(samdf) <- samdf$sample_name

# # import taxa info
# taxa <- read.csv("tvm_symportal_taxa.csv",header=TRUE)
# rownames(taxa) <- as.factor(taxa$ITS2_type_profile)
# mtaxa <- as.matrix(taxa)

# import counts (absolute abundance from its2 type profiles)
mcounts <- as.matrix(counts)

# Construct phyloseq object 
ps.med <- phyloseq(otu_table(mcounts, taxa_are_rows=FALSE),
               sample_data(samdf))

ps.med
# phyloseq-class experiment-level object
# otu_table()   OTU Table:         [ 336 taxa and 41 samples ]
# sample_data() Sample Data:       [ 41 samples by 7 sample variables ]

#saveRDS(ps.med,file="ps.its2.postmed.rds")
```

## Plotting

```{r}
ps.med.ord <- ordinate(ps.med,"PCoA",distance="bray")

plot_ordination(ps.med, ps.med.ord, color ="timepoint")+
  geom_point(alpha=0.5)+
  #scale_color_manual(values=c("darkslategray3","darkslategray4","#000004"))+
  #scale_shape_manual(values=c(16,15))+
  stat_ellipse()+
  facet_grid(treatment~host_species)+
  theme_cowplot()
```

## Statz

```{r}
seq.med <- data.frame(ps.med@otu_table)
samdf.med <- data.frame(ps.med@sam_data)

dist.med <- vegdist(seq.med, method="bray")

bet.med <- betadisper(dist.med,samdf.med$treatment)
permutest(bet.med,permutations=999)
plot(bet.med) #ns
bet.med <- betadisper(dist.med,samdf.med$timepoint)
permutest(bet.med,permutations=999)
plot(bet.med) #ns

adonis2(dist.med ~ treatment*timepoint*host_species, data=samdf.med, permutations=999) #ns

##all
ps.med.fav <- subset_samples(ps.med,host_species=="faveolata")

seq.med.fav <- data.frame(ps.med.fav@otu_table)
samdf.med.fav <- data.frame(ps.med.fav@sam_data)

dist.med.fav <- vegdist(seq.med.fav, method="bray")

bet.med.fav <- betadisper(dist.med.fav,samdf.med.fav$treatment)
permutest(bet.med.fav,permutations=999)
plot(bet.med.fav) #ns
bet.med.fav <- betadisper(dist.med.fav,samdf.med.fav$timepoint)
permutest(bet.med.fav,permutations=999)
plot(bet.med.fav) #ns

adonis2(dist.med.fav ~ treatment*timepoint, data=samdf.med.fav, permutations=999) #ns

##franks
ps.med.fra <- subset_samples(ps.med,host_species=="franksi")

seq.med.fra <- data.frame(ps.med.fra@otu_table)
samdf.med.fra <- data.frame(ps.med.fra@sam_data)

dist.med.fra <- vegdist(seq.med.fra, method="bray")

bet.med.fra <- betadisper(dist.med.fra,samdf.med.fra$treatment)
permutest(bet.med.fra,permutations=999)
plot(bet.med.fra) #ns
bet.med.fra <- betadisper(dist.med.fra,samdf.med.fra$timepoint)
permutest(bet.med.fra,permutations=999)
plot(bet.med.fra) #ns

adonis2(dist.med.fra ~ treatment*timepoint, data=samdf.med.fra, permutations=999) #ns
```
